src/Service/HealthCheck/ComprehensiveHealthReportResponseWriter.cs

```diff
@@ -62,32 +68,72 @@ public ComprehensiveHealthReportResponseWriter(
         /// </summary>
         /// <param name="context">HttpContext for writing the response.</param>
         /// <returns>Writes the http response to the http context.</returns>
-        public Task WriteResponse(HttpContext context)
+        public async Task WriteResponse(HttpContext context)
         {
             RuntimeConfig config = _runtimeConfigProvider.GetConfig();
 
             // Global comprehensive Health Check Enabled
-            if (config.IsHealthEnabled)
+            if (config != null && config.IsHealthEnabled)
             {
                 _healthCheckHelper.UpdateIncomingRoleHeader(context);
                 if (!_healthCheckHelper.IsUserAllowedToAccessHealthCheck(context, config.HostMode, config.AllowedRolesForHealth))
                 {
                     LogTrace("Comprehensive Health Check Report is not allowed: 403 Forbidden due to insufficient permissions.");
                     context.Response.StatusCode = StatusCodes.Status403Forbidden;
-                    return context.Response.CompleteAsync();
+                    await context.Response.CompleteAsync();
+                    return;
                 }
 
-                ComprehensiveHealthCheckReport dabHealthCheckReport = _healthCheckHelper.GetHealthCheckResponse(context, config);
-                string response = JsonSerializer.Serialize(dabHealthCheckReport, options: new JsonSerializerOptions { WriteIndented = true, DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull });
-                LogTrace($"Health check response writer writing status as: {dabHealthCheckReport.Status}");
-                return context.Response.WriteAsync(response);
+                string? response;
+                // Check if the cache is enabled 
+                if (config.CacheTtlSeconds != null && config.CacheTtlSeconds > 0)
+                {
+                    if (!_cache.TryGetValue(CACHE_KEY, out response))
+                    {
+                        ComprehensiveHealthCheckReport dabHealthCheckReport = _healthCheckHelper.GetHealthCheckResponse(context, config);
+
+                        response = JsonSerializer.Serialize(dabHealthCheckReport, options: new JsonSerializerOptions { WriteIndented = true, DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull });
+                        LogTrace($"Health check response writer writing status as: {dabHealthCheckReport.Status}");
```

Comment:
Do you think it would look better if it was all capitalized or linked with `-`?
```suggestion
                        LogTrace($"Health Check Response Writer writing status as: {dabHealthCheckReport.Status}");
```

Reply:
i think capitalized or not it should be okay since we are not quoting a particular file.